ArchUnitNET/Domain/PlantUml/ComponentName.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace ArchUnitNET.Domain.PlantUml
+{
+    internal class ComponentName
+    {
+        private readonly string _value;
+
+        public ComponentName(string value)
+        {
+            _value = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        public string AsString()
+        {
+            return _value;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is ComponentName name &&
+                   _value == name._value;
+        }
+
+        public override int GetHashCode()
+        {
+            return -1939223833 + EqualityComparer<string>.Default.GetHashCode(_value);
```

Comment:
how did you generate the equals/gethashcode methods? would it be possible to use the convention we have in other places?

Reply:
Visual Studio 2019 generated that code. I will have a look at other equals methods and use that convention.

Reply:
I can also do this. I assume we used Rider/Resharper